### ISSUE TITLE
Questions about Contrastive Loss Function

### ISSUE THREAD

**Issue by mxqmxqmxq** [yuyangw/MolCLR#30]
Great work, simple and efficient! However, I have some doubts about the implementation of the NTXentLoss function. Why is the shape of logits 1024, 1023, while the shape of the labels is 1024? After that, why is a torch.nn.CrossEntropyLoss used? Why is this done this way? Isn't the general implementation simply -log(loss_positive / loss_negative)?
![image](https://github.com/user-attachments/assets/daad8c4d-eaec-4ab5-abe9-417dc9b55925)
